### ISSUE TITLE
[Enhancement] More significant figures on "Hit Chance"

### ISSUE THREAD

**Issue by aggixx** [Openarl/PathOfBuilding#422]
Currently the "Hit Chance" stat is rounded to a whole percent. This causes stats like Dex, flat accuracy, % accuracy, etc to have extremely granularity in their DPS value. For example, on the character I am looking at right now there is no difference whatsoever between a minimum T1 accuracy roll and a maximum one, this obscures the "real" value of accuracy rating on that item and the same thing can be said about accuracy nodes on the passive tree.

**Comment by Openarl** [comment]
The Hit Chance in Path of Building is rounded for two reasons:
1. Consistency with the game's output; indeed all the program's outputs are rounded the same way as the game's output, where applicable. As for why the game rounds it that way, it's hard to know for sure, but it's probably because...
2. The Hit Chance calculation is by necessity an estimate, since your actual Hit Chance depends on the enemy's evasion, which varies considerably, and so therefore does the value of Accuracy. Consequently, attempting to min-max your Accuracy is pretty pointless, and rounding the hit chance is a pretty handy way of discouraging the user from attempting to do so.
I know that min-maxing every little detail of a build is the main point of Path of Building, but where Accuracy is concerned all you can really do is get "enough", then build everything else around that. It sucks, but that's just the way the game mechanics are.

**Comment by aggixx** [comment]
I agree that the hit chance listed in the stats pane on the left should be displayed in the same way as it is displayed in game, but the DPS and crit chance calculations actually use that rounded value when (as far as I know) the game engine itself only rounds the value for display on the character sheet.

> The Hit Chance calculation is by necessity an estimate, since your actual Hit Chance depends on the enemy's evasion, which varies considerably, and so therefore does the value of Accuracy. Consequently, attempting to min-max your Accuracy is pretty pointless, and rounding the hit chance is a pretty handy way of discouraging the user from attempting to do so.

I disagree. While the value of accuracy in PoB is certainly an approximate one, you are making that estimate even more useless by arbitrarily rounding it in the damage calculations. The fact that I can put on an item with 100 flat accuracy and see no DPS increase whatsoever even though in-game that item is an objective improvement is rather silly.

**Comment by aggixx** [comment]
> Accuracy is concerned all you can really do is get "enough", then build everything else around that. 

How am I supposed to know what "enough" is when the displayed value of accuracy is almost so inaccurate as to be useless if you have any considerable amount of it.

**Comment by Openarl** [comment]
> I disagree. While the value of accuracy in PoB is certainly an approximate one, you are making that estimate even more useless by arbitrarily rounding it in the damage calculations. The fact that I can put on an item with 100 flat accuracy and see no DPS increase whatsoever even though in-game that item is an objective improvement is rather silly.

The alternative is the program showing a DPS increase that isn't an accurate representation of how much that extra accuracy will affect your DPS. Also, any amount of accuracy that _doesn't_ increase the displayed DPS would increase your actual DPS by less than 1% (2% if you're a crit build), which isn't usually worth the trouble.
> How am I supposed to know what "enough" is when the displayed value of accuracy is almost so inaccurate as to be useless if you have any considerable amount of it.

"Enough" is however much gives you an acceptable hit chance. 90% is usually a good target; since accuracy has diminishing returns there's generally little benefit from going further. Or you can just keep getting accuracy until the amounts you're adding aren't consistently increasing the hit chance, as that's a good sign that you need to stop.

**Comment by tanis0** [comment]
Hi Openarl. I love PoB, but please reconsider this request.

I can see why you might not want to encourage Earthquake builds to min-max accuracy, but it can be more relevant to other builds. For example, I'm building a 13+ attack rate crit frost-blade character. It will take several hundred hits to kill end-game bosses, so effective hit chance and effective crit chance should have relatively predictable effects there. I have a choice between a 4% attack speed node or an 18% accuracy node and if I'm doing my napkin math correctly, the accuracy node is better even when prioritizing accuracy elsewhere in the tree and getting some on gear (PoB shows 90% hit rate). However, PoB shows no increase for that node vs a 1.2% DPS increase for the attack speed node. Additionally, the value per node of clusters such as Depth Perception is harder to predict than it should be as the DPS numbers can't be trusted.

Also, while it's true that you need some idea of enemy evasion in order to predict the value of accuracy, I don't see how this is different from needing some idea of enemy resists to accurately predict the value of penetration. Why not handle accuracy in a similar way?

My suggestion is to allow more granularity on hit chance, to add a box to the Configuration menu to override enemy evasion, and to add a configuration dropdown or checkbox to opt in to the new behavior if you want to leave the current behavior as the default.
